tests/test_PessimisticPatternMatcherB3D.py

```diff
@@ -0,0 +1,273 @@
+        self.source_obj_array = copy(self.reference_obj_array)
+        self.log = Log()
+
+    def testConstructPattern(self):
+        """ Test that a specified pattern can be found in the refernce
+        data and that the explicit ids match.
+        """
+        self.pyPPMb = PessimisticPatternMatcherB(
+            reference_catalog=self.reference_obj_array[:, :3],
+            log=self.log)
+
+        pattern_struct = self.pyPPMb._construct_pattern_and_shift_rot_matrix(
+            self.source_obj_array[:6, :3], 6, np.cos(np.radians(60. / 3600.)),
+            np.cos(np.radians(1.0)) ** 2, np.radians(5./3600.))
+        pattern_list = pattern_struct.ref_candidates
+        self.assertGreater(len(pattern_list), 0)
+        self.assertEqual(pattern_list[0], 0)
+        self.assertEqual(pattern_list[1], 1)
+        self.assertEqual(pattern_list[2], 2)
+        self.assertEqual(pattern_list[3], 3)
+        self.assertEqual(pattern_list[4], 4)
+        self.assertEqual(pattern_list[5], 5)
+
+        pattern_struct = self.pyPPMb._construct_pattern_and_shift_rot_matrix(
+            self.source_obj_array[:9, :3], 6, np.cos(np.radians(60. / 3600.)),
+            np.cos(np.radians(1.0)) ** 2, np.radians(5./3600.))
+        pattern_list = pattern_struct.ref_candidates
+        self.assertGreater(len(pattern_list), 0)
+        self.assertEqual(pattern_list[0], 0)
+        self.assertEqual(pattern_list[1], 1)
+        self.assertEqual(pattern_list[2], 2)
+        self.assertEqual(pattern_list[3], 3)
+        self.assertEqual(pattern_list[4], 4)
+        self.assertEqual(pattern_list[5], 5)
+
+        pattern_struct = self.pyPPMb._construct_pattern_and_shift_rot_matrix(
+            self.source_obj_array[[2, 4, 8, 16, 32, 64], :3], 6,
+            np.cos(np.radians(60. / 3600.)), np.cos(np.radians(1.0)) ** 2,
+            np.radians(5./3600.))
+        pattern_list = pattern_struct.ref_candidates
+        self.assertEqual(pattern_list[0], 2)
+        self.assertEqual(pattern_list[1], 4)
+        self.assertEqual(pattern_list[2], 8)
+        self.assertEqual(pattern_list[3], 16)
+        self.assertEqual(pattern_list[4], 32)
+        self.assertEqual(pattern_list[5], 64)
+
+    def testMatchPerfect(self):
+        """ Input objects that have no shift or rotation to the matcher
+        and test that we return a match.
+        """
+        self.pyPPMb = PessimisticPatternMatcherB(
+            reference_catalog=self.reference_obj_array[:, :3],
+            log=self.log)
+
+        match_struct = self.pyPPMb.match(
+            source_catalog=self.source_obj_array, n_check=9, n_match=6,
+            n_agree=2, max_n_patterns=100, max_shift=60., max_rotation=5.0,
+            max_dist=5., min_matches=30, pattern_skip_array=None)
+        self.assertEqual(len(match_struct.matches),
+                         len(self.reference_obj_array))
+        self.assertTrue(
+            np.all(match_struct.distances < 0.01/3600.0 * __deg_to_rad__))
```

Comment:
Consider using numpy.testing.assert_all; also please use named constants for clarity (here and elsewhere). For example:

```
from numpy.testing import assert_all
...
maxAllowedDistRad = 0.01/3600.0 * __deg_to_rad__
assert_all(match_struct.distances < maxAllowedDistRad)
```